orm/hibernate-orm-4/src/test/java/org/hibernate/bugs/ORMStandaloneJpaUnitTestCase.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2014 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hibernate.bugs;
+
+import org.hibernate.cfg.Environment;
+import org.hibernate.internal.util.StringHelper;
+import org.hibernate.jpa.AvailableSettings;
+import org.hibernate.jpa.HibernatePersistenceProvider;
+import org.hibernate.jpa.boot.spi.Bootstrap;
+import org.hibernate.jpa.boot.spi.PersistenceUnitDescriptor;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.SharedCacheMode;
+import javax.persistence.ValidationMode;
+import javax.persistence.spi.PersistenceUnitTransactionType;
+import java.net.URL;
+import java.util.*;
+
+/**
+ * This template demonstrates how to develop a standalone test case for Hibernate ORM using the JPA specification.
+ */
+public class ORMStandaloneJpaUnitTestCase {
+
+	private EntityManagerFactory entityManagerFactory;
+
+    // Add your entities here.
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[0];
+	}
+
+    // If you use *.hbm.xml mappings, instead of annotations, add the mappings here.
+    protected String[] getMappings() {
+        return new String[0];
+    }
+
+	// Add your tests, using standard JUnit.
+	@Test
+	public void hhh123Test() throws Exception {
+
+	}
+
+    @Before
+    public void setup() {
+        entityManagerFactory =  Bootstrap.getEntityManagerFactoryBuilder(
+                buildPersistenceUnitDescriptor(),
+                buildSettings()
+        ).build();
+    }
+
+	private PersistenceUnitDescriptor buildPersistenceUnitDescriptor() {
```

Comment:
In my opinion, I would never go for the standalone test and just override a base class that offers all that just like for the native test cases. Since, that's not possible for the hibernate-eneitytmanager, which doesn't publishes a test jar as well and because hibernate-testing is not based on JPA, we can supply it separately here.

For the native test cases, we advise people to use the `ORMUnitTestCase.java` and not the standalone one. We could do the same for JPA too, even if we supply this class in this project and we don't inherit it from other module.